docs/en/guides/developer/insertion-deduplication-on-retries.md

```diff
@@ -0,0 +1,390 @@
+
+Each time that data is inserted to the destination table, `block_id` is written to the deduplication log. For each insertion the condition is checked that there is no such `block_id` in deduplication log. If `block_id` is found in deduplication log, than the block is considered as duplicate. Note that Deduplication log stores finite count of `block_id`'s. Only insertions which meet the deduplication log window parameters have chances to be deduplicated.
+
+For `INSERT VALUES` queries splitting the inserted data to the block is deterministic and it is determined by settings. Therefore user shoud retry insertions with the same settings values as they were at first operation.
+
+For `INSERT SELECT` queries it is important that `SELECT` part of the query returns the same data in the same order each try. That is hard to achive in practical usage for many reasons. In order to achieve the stable data order on retries you could define precise `ORDER BY` section in `SELECT` part of the query. But the selected table could be updated between retries therefore the result data could change. Also there are could be a lot of data as a result a lot of blocks after sptitting inserted data by rows count and bytes count. That count of blocks might overflow deduplication log window.
+
+
+# How insertion deduplication works with materialized views?
+When a table has one or several materialized views, the inserted data is also inserted to the destination of that views with some defined transformation.
+That transformed data also has to be deduplicated on retries. Clickhouse deduplicates it in the same way it deduplicates data which is inserted into target table.
+User could control that process with settings on the table under materialized view: `replicated_deduplication_window`, `replicated_deduplication_window_seconds` and `non_replicated_deduplication_window`. Also user could use profile setting `deduplicate_blocks_in_dependent_materialized_views`.
+For the blocks inserted in tables under materialized views Clickhouse calulates `block_id` as hash from a string which contains concatenation of `block_id`'s for the the source table and other parts which helps to distiguish blocks afret materialised views transformation, like source view's id and the sequential number of that block. That makes deduplication for materialised views works correctly and distinguish the data by its original inserted data, no mater how it has been transormed on its way to the destination table under materialized view.
+
```

Comment:
Also mention that `update_insert_deduplication_token_in_dependent_materialized_views` is deprecated after some version?